docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md

```diff
@@ -0,0 +1,496 @@
+
+  **Command**
+
+  ```
+  $ kubectl create ns csi-nfs
+  ```
+
+  **Output**
+
+  ```
+  namespace/csi-nfs created
+  ```
+
+   - Install NFS CSI Driver using helm.
+
+  **Command**
+
+  ```
+  $ helm install -n csi-nfs  csi-nfs
+  ```
+  
+  **Output**
+
+  ```
+  NAME: csi-nfs
+  LAST DEPLOYED: Wed Jun 19 15:39:25 2024
+  NAMESPACE: csi-nfs
+  STATUS: deployed
+  REVISION: 1
+  TEST SUITE: None
+  NOTES:
+  The CSI NFS Driver is getting deployed to your cluster.
+  ```
+
+   - Verify the CSI NFS Driver pods status.
+  
+  ```
+  kubectl --namespace=csi-nfs get pods --selector="app.kubernetes.io/instance=csi-nfs" --watch
+  ```
+
+   - Verify the pod status.
+
+  **Command**
+
+  ```
+  $ kubectl get pods -n csi-nfs 
+  ```
+
+  **Output**
+
+  ```
+  NAME                                 READY   STATUS    RESTARTS   AGE
+  csi-nfs-controller-7fff64574-cnwdp   4/4     Running   0          53s
+  csi-nfs-node-j27w7                   3/3     Running   0          53s
+  csi-nfs-node-slp95                   3/3     Running   0          53s
+  csi-nfs-node-srsbm                   3/3     Running   0          53s
+  ```
+
+   - Verify the storage class.
+
+  **Command**
+  
+  ```
+  $ kubectl get sc | grep nfs-csi
+  ```
+
+  **Output**
+
+  ```
+  nfs-csi                   nfs.csi.k8s.io            Delete          Immediate              false                  2m29s
+  ```
+
+  **Command**
+  
+  ```
+  $ kubectl describe sc nfs-csi
+  ```
+
+  **Output**
+
+  ```
+  Name:                  nfs-csi
+  IsDefaultClass:        No
+  Annotations:           meta.helm.sh/release-name=csi-nfs,meta.helm.sh/release-namespace=csi-nfs
+  Provisioner:           nfs.csi.k8s.io
+  Parameters:            mountPermissions=0,server=nfs-server.nfs-server.svc.cluster.local,share=/
+  AllowVolumeExpansion:  <unset>
+  MountOptions:
+    nfsvers=4.1
+  ReclaimPolicy:      Delete
+  VolumeBindingMode:  Immediate
+  Events:             <none>
+  ```
+
+### Provisioning RWX PVCs
+
+1. Create PVC with NFS CSI Driver Storage Class `nfs-csi` (that was created in last step).
+
+**YAML**
+
+```
+$ cat nfs-pvc.yaml 
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: nfs-claim-1
+spec:
+  storageClassName: nfs-csi
+  accessModes: [ "ReadWriteMany" ]
+  resources:
+    requests:
+       # This is the storage capacity that will be available to the application pods.
+      storage: 5Gi
+```
+
+**Command**
+
+```
+$ kubectl apply -f nfs-pvc.yaml 
+```
+
+**Output**
+
+```
+persistentvolumeclaim/nfs-claim-1 created
+```
+
+- Use the following command to verify the PVC:
+
+**Command**
+
+```
+$ kubectl get pvc 
+```
+
+**Output**
+
+```
+NAME          STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
+nfs-claim-1   Bound    pvc-f795f8bc-58b3-40d7-b9b2-2d16c17f3d47   5Gi        RWX            nfs-csi        <unset>                 10s
+```
+
+2. Deploy the application.
```

Comment:
Is nginx the right example here, @datacore-tilangovan ?
I think, we also need to show-case rwx use-case here, by deploying more than one pod/application using the shared storage.

Reply:
we have nginx with three replicas here using pvc (nfs-claim-1).

we can use standalone Db's as example, but keeping them replicated as deployments is not ideal use case as they will be deployed as statefulsets. so we have used nginx here.

Reply:
ok